patches/server/1047-Configurable-memory-condition-tick-rate.patch

```diff
@@ -0,0 +1,30 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: ruViolence <78062896+ruViolence@users.noreply.github.com>
+Date: Wed, 8 Nov 2023 01:25:45 +0800
+Subject: [PATCH] Configurable memory condition tick rate
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java b/src/main/java/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java
+index 6f323026b2d6fb5151e8b9dd9466e96eb026ccbd..6aeb432d7e79cbf5c64da934bd6d4d63a7a35e44 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java
++++ b/src/main/java/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java
+@@ -333,8 +333,19 @@ public class BehaviorBuilder<E extends LivingEntity, M> implements App<BehaviorB
+     static final class PureMemory<E extends LivingEntity, F extends K1, Value> extends BehaviorBuilder<E, MemoryAccessor<F, Value>> {
+         PureMemory(final MemoryCondition<F, Value> query) {
+             super(new BehaviorBuilder.TriggerWithResult<E, MemoryAccessor<F, Value>>() {
++                // Paper start - memory condition tick rate
++                private final String configKey = query.memory().toString().toLowerCase(java.util.Locale.ROOT);
```

Comment:
Ok, looking at the impl here, I don't really like using the toString() method here. Just call the getKey() method on `BuiltInRegistries.MEMORY_MODULE_TYPE` directly. Best to toString the ResourceLocation directly as that has less chance of changing in the future.